### PR TITLE
follow changes in Modal

### PR DESCRIPTION
**Pull request by jsipos**
- Modal has deprecated the `secret` parameter in `stub.function`: now `secrets` should be used instead. Updated usages.
- Make the `make_model_era5_template` call explicitly `.local()`

### CODE REVIEW

File: ai-models-modal/main.py [hunk 1]
```diff
@@ -439,7 +439,7 @@ def _maybe_download_assets(model_name: str) -> None:
         )
         if not target_blob.exists():
             logger.info("  Template not found; generating from scratch.")
-            make_model_era5_template(model_name)
+            make_model_era5_template.remote(model_name)
```
Comment (darothen) [inline ai-models-modal/main.py]: This is deliberate; the call stack here (which is opaque; needs to be refactored) starts with generate_forecast(), which is cheap to run (we don't mount a GPU). So the intention is that we can run the template generation locally without too much hassle. The template generation function is wrapped as a standalone function to make it easier to test or run ad hoc.

WDYT? I don't see a big reason this needs to be a remote() call, but perhaps we should add a TODO indicating that the entire process should be re-factored?

Comment (jsipos) [reply ai-models-modal/main.py]: My goal with adding .remote here was to get rid of a warning from Modal saying that either .local or .remote should be used. In light of this being a cheap call, I'd have opted for .local instead. However in the current version I'm not even getting this warning any more, so I think I'll just remove this change, and let the future refactoring take care of it. Do you agree?

Comment (darothen) [reply ai-models-modal/main.py]: SGTM! Thanks for playing with this.

Comment (jsipos) [reply ai-models-modal/main.py]: Updated the call to an explicit .local(), I think it should be fine like this.

**Comment by darothen** [comment]
LGTM

**Pull request merged**